spec/emerald_spec.rb

```diff
@@ -0,0 +1,129 @@
+require 'spec_helper'
+require 'tempfile'
+require 'helper/user_comms'
+include Emerald
+require 'emerald/parser'
+require 'emerald/compiler'
+
+
+RSpec.describe 'Emerald' do
+
+  let(:stdout) { spy('STDOUT')}
+  let(:user_comms) { UserCommsHelper.new(stdout) }
+
+  describe "given a non-lisp file is passed in" do
+    context "when non-lisp file is passed" do
+      it "returns an InvalidFileType error" do
+        expect{ user_comms.verify_lisp_file('test-file.rb') }.to raise_error(Emerald::UserCommsHelper::InvalidFileTypeError).with_message(Emerald::UserCommsHelper::ERROR_INCORRECT_FILE_TYPE)
+      end
+    end
+
+    context "when no file is passed" do
+      it "returns an InvalidFileType error" do
+        expect{ user_comms.verify_lisp_file('') }.to raise_error(Emerald::UserCommsHelper::InvalidFileTypeError).with_message(Emerald::UserCommsHelper::ERROR_INCORRECT_FILE_TYPE)
+      end
+    end
+  end
+
+  describe "given a valid lisp file is passed" do
+    let(:test_file) { Tempfile.new('new-file.lisp') }
+
+    context "when it receives an invalid list without a closing bracket" do
+      it "raises an InvalidListError" do
+        test_file.write('(+ 5 2')
+        test_file.close
+
+        allow(user_comms).to receive(:new).with(stdout)
+        allow(user_comms).to receive(:verify_lisp_file).and_return(test_file)
+
+        lisp_file_contents = File.read(test_file)
```

Comment:
Yes, you're right. The stubs are unnecessary, I think my reasoning here at the time was to create a way of guiding the tests to this error state scenario, but obviously the stubs aren't actually serving a purpose